android/delegate.gradle

```diff
@@ -0,0 +1,153 @@
+buildscript {
+    repositories {
+        mavenCentral()
+        mavenLocal()
+    }
+    dependencies {
+        classpath 'com.android.tools.build:gradle:1.3.+'
+        classpath 'com.jakewharton.sdkmanager:gradle-plugin:0.12.+'
+        classpath 'com.github.dcendents:android-maven-plugin:1.2'
+    }
+}
+apply plugin: 'android-sdk-manager'
+apply plugin: 'com.android.library'
+apply plugin: 'com.github.dcendents.android-maven'
+apply plugin: 'signing'
+
+def properties = new Properties()
+properties.load(new FileInputStream(file('../version.properties')))
+
+description = 'gRPC: Android'
+group = "io.grpc"
+version = "${properties.version}"
+
+repositories {
+    mavenCentral()
+    mavenLocal()
+}
+
+project.archivesBaseName = "grpc-android"
+
+android {
+    compileSdkVersion 21
+    buildToolsVersion '21.1.1'
+
+    defaultConfig {
+        minSdkVersion 9
+        targetSdkVersion 21
+        versionCode 1
+        versionName "1.0"
+    }
+    lintOptions {
+        disable 'InvalidPackage', 'HardcodedText'
+    }
+}
+
+// Disable JavaDoc doclint on Java 8. It's annoying.
+if (JavaVersion.current().isJava8Compatible()) {
+    allprojects {
+        tasks.withType(Javadoc) {
+            options.addStringOption('Xdoclint:none', '-quiet')
+        }
+    }
+}
+
+configurations {
+    javadocDeps
+}
+
+dependencies {
+    compile "io.grpc:grpc-okhttp:${properties.version}"
+    javadocDeps "io.grpc:grpc-okhttp:${properties.version}"
+
+    testCompile 'junit:junit:4.12'
+    testCompile 'org.mockito:mockito-core:1.9.5'
+}
+
+signing {
+    required false
+    sign configurations.archives
+}
+
+// Must match uploadTask.repositories.mavenDeployer.pom.artifactId
+
+//archiveTask.baseName = 'grpc-android'
+
+
+uploadArchives.repositories.mavenDeployer {
+    beforeDeployment { MavenDeployment deployment -> signing.signPom(deployment) }
+    String stagingUrl
+    if (rootProject.hasProperty('repositoryId')) {
+        stagingUrl = 'https://oss.sonatype.org/service/local/staging/deployByRepositoryId/' +
+                rootProject.repositoryId
+    } else {
+        stagingUrl = 'https://oss.sonatype.org/service/local/staging/deploy/maven2/'
+    }
+    def configureAuth = {
+        if (rootProject.hasProperty('ossrhUsername') && rootProject.hasProperty('ossrhPassword')) {
+            authentication(userName: rootProject.ossrhUsername, password: rootProject.ossrhPassword)
+        }
+    }
+    repository(url: stagingUrl, configureAuth)
+    snapshotRepository(url: 'https://oss.sonatype.org/content/repositories/snapshots/', configureAuth)
+    pom.artifactId = 'grpc-android'
```

Comment:
Unnecessary? since `When you set the “archiveTask.baseName” property to a value other than the default, you'll also have to set uploadTask.repositories.mavenDeployer.pom.artifactId to the same value` and `archiveTask.baseName defaults to project.archivesBaseName which in turn defaults to project.name`